vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -92,8 +118,18 @@ def update_authentication(data, options = {})
       end
 
       # Update or create
-      cred = self.authentications.build(:name => "#{self.class.name} #{self.name}", :authtype => type.to_s, :type => "AuthUseridPassword") if cred.nil?
-      cred.userid, cred.password = value[:userid], value[:password]
+      if cred.nil?
+        if self.kind_of?(EmsOpenstackInfra) && value[:auth_key]
```

Comment:
Maybe not a builder but maybe we can have a method on the `EmsOpenstackInfra` instance so we don't need this conditional.  it's not required right now but something to keep in mind.